extensions/broadcast_channel/lib.rs

```diff
@@ -0,0 +1,163 @@
+// Copyright 2018-2021 the Deno authors. All rights reserved. MIT license.
+
+use async_trait::async_trait;
+use deno_core::error::bad_resource_id;
+use deno_core::error::AnyError;
+use deno_core::include_js_files;
+use deno_core::op_async;
+use deno_core::op_sync;
+use deno_core::Extension;
+use deno_core::OpState;
+use deno_core::Resource;
+use deno_core::ResourceId;
+use deno_core::ZeroCopyBuf;
+use std::cell::RefCell;
+use std::path::PathBuf;
+use std::rc::Rc;
+use std::sync::atomic::AtomicU64;
+use std::sync::atomic::Ordering::Relaxed;
+use std::sync::Arc;
+use std::sync::Mutex;
+use tokio::sync::broadcast;
+use tokio::sync::broadcast::Receiver;
+use tokio::sync::broadcast::Sender;
+
+#[async_trait]
+pub trait BroadcastChannel {
+  type Resource: Resource;
+
+  fn open(name: String) -> Result<Self::Resource, AnyError>;
+
+  async fn send(
+    resource: &Self::Resource,
+    data: Vec<u8>,
+  ) -> Result<(), AnyError>;
+
+  async fn next_event(
+    resource: &Self::Resource,
+  ) -> Result<Option<Vec<u8>>, AnyError>;
+}
+
+pub struct InMemoryBroadcastChannel;
+
+#[async_trait]
+impl BroadcastChannel for InMemoryBroadcastChannel {
+  type Resource = InMemoryBroadcastChannelResource;
+
+  fn open(name: String) -> Result<Self::Resource, AnyError> {
+    static ID: AtomicU64 = AtomicU64::new(0);
+    lazy_static::lazy_static! {
+      static ref SENDER: Mutex<Sender<Event>> =
+        Mutex::new(broadcast::channel(64).0);
+    }
```

Comment:
I have some reservations about this. This state should be stored in OpState / ProcessState, just like BlobURLStore. If it is just process global, it will break in all kinds of ways in `deno test`

Reply:
You mean ProgramState? I'll look into that.